### PR TITLE
Collab backports

### PR DESCRIPTION
**Pull request by Kelketek**
Pulls the line editor into the main area, allows it to be overridden, adds long desc option, and also adds an exit to the test scenarios.

### CODE REVIEW

File: evennia/commands/default/building.py [hunk 1]
```diff
@@ -515,12 +515,18 @@ class CmdDesc(MuxCommand):
...
+    Usage 1:
       @desc [<obj> =] >description>
 
-    Setts the "desc" attribute on an
+    Sets the "desc" attribute on an
     object. If an object is not given,
     describe the current room.
+
+    Usage 2:
+      @desc/long [obj]
```
Comment (Griatch) [inline evennia/commands/default/building.py]: This seems to refer to "/long" still, whereas the switch is /edit in the code below.

File: evennia/commands/default/building.py [hunk 2]
```diff
@@ -531,22 +537,60 @@ def func(self):
...
-        else:
+            obj.db.desc = self.rhs
+            desc_set = True
+        elif 'edit' not in self.switches:
             obj = caller.location
-            desc = self.args
-        # storing the description
-        obj.db.desc = desc
-        caller.msg("The description was set on %s." % obj.key)
+            obj.db.desc = self.args
```
Comment (Griatch) [inline evennia/commands/default/building.py]: This seems like a rather hard-to-read way to do it. Why not just an if ... else statement for the editor or non-editor cases?

File: evennia/utils/utils.py [hunk 3]
```diff
@@ -1257,3 +1257,10 @@ def m_len(target):
     if inherits_from(target, basestring):
         return len(ANSI_PARSER.strip_mxp(target))
     return len(target)
+
+
+def get_line_editor():
```
Comment (Griatch) [inline evennia/utils/utils.py]: Nice - customizable line editor location. :)

Comment (Kelketek) [inline evennia/utils/utils.py]: Seemed to me we'd better have it if commands were expected to depend on it. :)

**Pull request merged**